### PR TITLE
Bump github.com/tetratelabs/wazero from 1.0.0-pre.7 to 1.0.0-pre.9 in /sdks

### LINKED ISSUE #25429: [Task]: Fix github.com/tetratelabs/wazero from 1.0.0-pre.7 to 1.0.0-pre.8 in /sdks
### What needs to happen?

PR #25249 is failing because wazero made a breaking change to their API:
```
examples/wasm/wasm.go:118:20: too many arguments in call to fn.r.NewHostModuleBuilder("env").NewFunctionBuilder().WithFunc(logString).Export("log").Instantiate
 	have (context.Context, wazero.Runtime)
 	want (context.Context)
```

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [X] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [X] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by damondouglas**
This PR fixes #25429 bumping wazero version to 1.0.0-pre.9 and refactoring the wasm example.  The fix was tested following steps documented in [sdks/go/examples/wasm/README](https://github.com/damondouglas/beam/tree/25429-fix-wazero-example/sdks/go/examples/wasm/README.md) on the local, flink, and dataflow runners.  In all three environments wrote expected output:

```
Hello, Ada!
Hello, Lovelace!
Hello, World!
Hello, Beam!
Hello, Senior López!
Hello, Random unicorn emoji 🦄!
```

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - ~Update `CHANGES.md` with noteworthy changes.~
 - ~If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).~

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/go/examples/wasm/wasm.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func main() {
 func run(ctx context.Context) error {
 	p, s := beam.NewPipelineWithRoot()
 
-	in := beam.Create(s, "Ada", "Lovelace", "World", "Beam", "Senior López")
+	in := beam.Create(s, "Ada", "Lovelace", "World", "Beam", "Senior López", "Random unicorn emoji 🦄")
```
Comment (lostluck) [inline sdks/go/examples/wasm/wasm.go:84]: 🦄

**Review by lostluck — CHANGES_REQUESTED** [review]
LGTM thanks!

**Review by lostluck — APPROVED** [review]

**Comment by damondouglas** [comment]
@lostluck FYI I added myself to watch https://github.com/tetratelabs/wazero for new releases to stay on top of this.

**Comment by lostluck** [comment]
Sounds good! Dependabot will also let someone know if another breaking change comes along. If we didn't care, we wouldn't be running the build/unit tests!

**Pull request merged**